### PR TITLE
Package secp256k1 JNI bindings

### PR DESCRIPTION
**Pull request by sstone**
We use a custom Secp256k1Context to package JNI bindings for 

- Linux 64 bits
- Windows 64 bits
- Osx 64 bits

Packaging is heavily inspired by the xerial sqlite-jdbc driver: dynamic libraries are embedded as resources in the JAR, and extracted and loaded from a temporary directory at runtime. 

Android developers: it does not seem to be possible to use embedded non-java JAR resources in this way, so you will still need to compile the secp256k1 JNI bindings in your application.

### CODE REVIEW

**Comment by pm47** [comment]
How can we tell bitcoin-lib to *not* use libsecp256k1 and use bouncycastle instead?

**Comment by Christewart** [comment]
Have you seen this native loader library? https://github.com/scijava/native-lib-loader 

We use it in bitcoin-s, not sure about android compatibility but it is pretty nice. 

https://github.com/bitcoin-s/bitcoin-s-core/tree/master/secp256k1jni

**Comment by sstone** [comment]
@pm47 to not use the JNI bindings just specify a path or name that does not exist (for example `-Dfr.acinq.secp256k1.lib.name=foo`)

**Comment by sstone** [comment]
@Christewart We use the loader from xerial-jdbc which does the same thing AFAICT. I chose it because it's a very popular project used on many platforms.

File: pom.xml [hunk 1]
```diff
@@ -145,6 +158,12 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-source-plugin</artifactId>
                 <version>3.0.1</version>
+                <configuration>
+                    <excludes>
+                        <!-- we also need to include the original Secp256k1Context.java from our source bundle -->
```
Comment (pm47) [inline pom.xml]:
```suggestion
                        <!-- we also need to exclude the original Secp256k1Context.java from our source bundle -->
```

**Pull request merged**